server/src/views/frame.js

```diff
@@ -225,6 +275,37 @@ class Frame extends React.Component {
     ga("send", "event", "website", "click-full-page-button", {page: location.toString()});
   }
 
+  onChange(e) {
+    let evt = new Event("hashchange");
+    window.dispatchEvent(evt);
+  }
+
+  onClickUploadFullPage(e) {
+    let uploadFullPageButton = document.getElementById("upload-full-page");
+    uploadFullPageButton.parentNode.removeChild(uploadFullPageButton);
+    let node = document.getElementById("full-page-checkbox");
+    node.checked = true;
+    let evt = new Event("hashchange");
+    window.dispatchEvent(evt);
+  }
+
+  onClickShareButton(e) {
+    let sharePanel = document.getElementById("share-buttons-panel");
+    console.log("SHARE STYLE", sharePanel.style.display);
+    if (sharePanel.style.display === "block") {
+      sharePanel.style.display = "none";
+    } else {
+      sharePanel.style.display = "block";
+    }
```

Comment:
This should be in state, I think?

Also clicking off the share panel should close the panel.  Probably the panel should be a subcomponent, and have its own hidden/showing state, with onComponentDidMount adding a handler to catch document click and hide the panel.